### ISSUE TITLE
Variable start timeout

### ISSUE THREAD

**Issue by Cooldude2606** [explosivegaming/factorio-olympics#50]
**Feature overview**
Name: Variable start timeout
TL;DR: The timeout while waiting for players should be dependent on the number of players remaning.

**Details**
Currently the timeout will last for 2 minutes regardless of the number of participants required to start the game, or the amount that are remaining. This should be changed so that the timeout does not start unless a minimum percentage is reach (maybe 50%) and then as less players are remaining the timeout is reduced.

**Possible methods**
Each time a participant is added or removed check_participant_count is called, during this call a new timeout can be created depending on the number of players and the start time. This will allow for fine control over the timeout without needing to check every tick for different condidtions to be met.
